dfelibs: new package

stephenswat opened the pull request:
This commits adds the `dfelibs` package, which is an Acts project fork of dfelibs.

stephenswat committed:
dfelibs: new package

This commits adds the `dfelibs` package, which is an Acts project fork
of dfelibs.

wdconinc commented on var/spack/repos/builtin/packages/acts-dfelibs/package.py:
```diff
@@ -0,0 +1,21 @@
+    list_url = "https://github.com/acts-project/dfelibs/tags"
+
+    maintainers("stephenswat")
+
```
```suggestion

    license("MIT", checked_by="stephenswat")

```

wdconinc commented on var/spack/repos/builtin/packages/acts-dfelibs/package.py:
```diff
@@ -0,0 +1,21 @@
+from spack.package import *
+
+
+class ActsDfelibs(CMakePackage):
```
```suggestion
class Dfelibs(CMakePackage):
```

We typically avoid naming packages differently for forks than for their upstream. I understand that upstream is unmaintained, but this avoids that someone else creates another `dfelibs` package independently of this one.

If at some point someone wants to add upstream (and cannot use the acts fork), or another fork, and if there really are conflicts between the patch sets that prevent merging into acts as a de facto maintained version, then we can always figure out something then.

wdconinc commented on var/spack/repos/builtin/packages/acts-dfelibs/package.py:
```diff
@@ -0,0 +1,21 @@
+
+    homepage = "https://github.com/acts-project/dfelibs"
+    url = "https://github.com/acts-project/dfelibs/archive/refs/tags/v20211029.tar.gz"
+    list_url = "https://github.com/acts-project/dfelibs/tags"
```
I'm curious: is this list_url necessary (again)?

Spack should be able to deduce the list URL for GitHub tag archives automatically, but occasionally GitHub changes something (including hiding html content behind dynamically loading JavaScript calls...) that breaks what we do.

stephenswat replied:
I guess not; I will remove it and test it.

stephenswat commented:
Thank you for the feedback @wdconinc; I have removed the `acts` from the name, added you as a maintainer, updated the license, and removed the unnecessary URL variable. I tested that the spec still installs! :+1:

wdconinc commented on var/spack/repos/builtin/packages/dfelibs/package.py:
```diff
@@ -0,0 +1,22 @@
+
+
+class Dfelibs(CMakePackage):
+    """Acts fork of dfelibs, a header-only untility library for C++."""
```
```suggestion
    """Acts fork of dfelibs, a header-only utility library for C++."""
```

wdconinc committed:
dfelibs: docstring typo

wdconinc reviewed (approved):
Thanks for adding this!
